hoot-services/src/main/java/hoot/services/command/ExternalCommandManagerImpl.java

```diff
@@ -0,0 +1,177 @@
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * --------------------------------------------------------------------
+ *
+ * The following copyright notices are generated automatically. If you
+ * have a new notice to add, please use the format:
+ * " * @copyright Copyright ..."
+ * This will properly maintain the copyright information. DigitalGlobe
+ * copyrights will be updated automatically.
+ *
+ * @copyright Copyright (C) 2016 DigitalGlobe (http://www.digitalglobe.com/)
+ */
+package hoot.services.command;
+
+import static hoot.services.HootProperties.CORE_SCRIPT_PATH;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+
+
+@Transactional
+@Component
+//@Profile("production")
+public class ExternalCommandManagerImpl implements ExternalCommandManager {
+    private static final Logger logger = LoggerFactory.getLogger(ExternalCommandManagerImpl.class);
+
+    // This contains the command runner objects for the executing commands. Used for job cancellation and tracking.
+    private static final Map<String, ExternalCommandRunner> activeCommands = new ConcurrentHashMap<>();
+
+    @Override
+    public String getProgress(String jobId) {
+        String stdStr = "";
+        ExternalCommandRunner commandRunner = activeCommands.get(jobId);
+        if (commandRunner != null) {
+            stdStr = commandRunner.getStdout();
+        }
+
+        return stdStr;
+    }
+
+    @Override
+    public void terminate(String jobId) {
+        ExternalCommandRunner cmdRunner = activeCommands.get(jobId);
+        if (cmdRunner != null) {
+            cmdRunner.terminate();
+        }
+    }
+
+    @Override
+    public CommandResult exec(String jobId, ExternalCommand command) {
+        String exec = command.get("exectype").toString();
+
+        String[] commandArr;
+        if (exec.equalsIgnoreCase("hoot")) {
+            commandArr = createCmdArray(command);
+        }
+        else if (exec.equalsIgnoreCase("make")) {
+            commandArr = createScriptCmdArray(command);
+        }
+        else if (exec.equalsIgnoreCase("bash")) {
+            commandArr = createBashScriptCmdArray(command);
+        }
```

Comment:
This if-else statement that checks for 'make', 'hoot', and 'bash' has been there for some time.  Once we decide on a more consistent approach of interfacing with the Core, I anticipate this code to be refactored.